include/fmt/std.h

```diff
@@ -445,61 +534,13 @@ struct formatter<
       return detail::write_bytes<Char>(out, string_view(ex.what()));
 
 #if FMT_USE_RTTI
-    const std::type_info& ti = typeid(ex);
-#  ifdef FMT_HAS_ABI_CXA_DEMANGLE
-    int status = 0;
-    std::size_t size = 0;
-    std::unique_ptr<char, void (*)(void*)> demangled_name_ptr(
-        abi::__cxa_demangle(ti.name(), nullptr, &size, &status), &std::free);
-
-    string_view demangled_name_view;
-    if (demangled_name_ptr) {
-      demangled_name_view = demangled_name_ptr.get();
-
-      // Normalization of stdlib inline namespace names.
-      // libc++ inline namespaces.
-      //  std::__1::*       -> std::*
-      //  std::__1::__fs::* -> std::*
-      // libstdc++ inline namespaces.
-      //  std::__cxx11::*             -> std::*
-      //  std::filesystem::__cxx11::* -> std::filesystem::*
-      if (demangled_name_view.starts_with("std::")) {
-        char* begin = demangled_name_ptr.get();
-        char* to = begin + 5;  // std::
-        for (char *from = to, *end = begin + demangled_name_view.size();
-             from < end;) {
-          // This is safe, because demangled_name is NUL-terminated.
-          if (from[0] == '_' && from[1] == '_') {
-            char* next = from + 1;
-            while (next < end && *next != ':') next++;
-            if (next[0] == ':' && next[1] == ':') {
-              from = next + 2;
-              continue;
-            }
-          }
-          *to++ = *from++;
-        }
-        demangled_name_view = {begin, detail::to_unsigned(to - begin)};
-      }
-    } else {
-      demangled_name_view = string_view(ti.name());
-    }
-    out = detail::write_bytes<Char>(out, demangled_name_view);
-#  elif FMT_MSC_VERSION
-    string_view demangled_name_view(ti.name());
-    if (demangled_name_view.starts_with("class "))
-      demangled_name_view.remove_prefix(6);
-    else if (demangled_name_view.starts_with("struct "))
-      demangled_name_view.remove_prefix(7);
-    out = detail::write_bytes<Char>(out, demangled_name_view);
-#  else
-    out = detail::write_bytes<Char>(out, string_view(ti.name())
-  });
-#  endif
+    out = detail::write_demangled_name<Char>(out, typeid(ex));
+#else
+    out = detail::write_bytes<Char>(out, string_view("std::exception"));
+#endif
```

Comment:
I am sorry :(
It was my mistake.
``#else`` code will never execute if ``FMT_USE_RTTI`` is disabled, because ``with_typename_ = FMT_USE_RTTI != 0;`` in ``parse()`` function.

More correct variant is remove this ``#else`` block and move ``#endif`` after ``return detail::write_bytes<Char>(out, string_view(ex.what()));``
This was before PR.

I'm sorry.

Reply:
Wouldn't it be more clear to do something like the below instead:
```c++
#if FMT_USE_RTTI
    if (with_typename_) {
      out = detail::write_demangled_name<Char>(out, typeid(ex));
      *out++ = ':';
      *out++ = ' ';
    }
#endif
    return detail::write_bytes<Char>(out, string_view(ex.what()));
```
?